### PR TITLE
Setup gitversion and pr template

### PR DESCRIPTION
**Pull request by jeanplevesque**
GitHub Issue: #

## Proposed Changes

 - [ ] Bug fix
 - [ ] Feature
 - [ ] Code style update (formatting)
 - [ ] Refactoring (no functional changes, no api changes)
 - [x] Build or CI related changes
 - [ ] Documentation content changes
 - [ ] Other, please describe:

## Description

- Changed the PR template.
- Replace the CHANGELOG with a simpler BREAKING_CHANGES file.
- Setup GitVersion configuration.
- Adapt pipeline for the master --> main branch rename.

## Impact on version

- [ ] **Major** (Public API was modified.)
  - Public constructs (class, struct, delegate, enum, etc.) were removed or renamed.
  - Public members were removed or renamed.
  - Public method signatures were changed or renamed.
- [ ] **Minor** (Public API was extended)
  - Public constructs, members, or overloads were added.
- [x] **Patch** (Public API was unchanged.)
  - A bug in behavior was changed.
  - Documentation was changed.

## Checklist

Please check if your PR fulfills the following requirements:

- [x] Documentation has been added/updated.
- [ ] Automated Unit / Integration tests for the changes have been added/updated.
- [ ] Updated [BREAKING_CHANGES.md](../BREAKING_CHANGES.md) (if you introduced a breaking change).
- [ ] Associated with an issue.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -3,40 +3,48 @@ GitHub Issue: #
...
 
-- [ ] Documentation has been added/updated
-- [ ] Automated Unit / Integration tests for the changes have been added/updated
-- [ ] Contains **NO** breaking changes
-- [ ] Updated the Changelog
-- [ ] Associated with an issue
+- [ ] Documentation has been added/updated.
+- [ ] Automated Unit / Integration tests for the changes have been added/updated.
+- [ ] Updated [BREAKING_CHANGES.md](../BREAKING_CHANGES.md) (if you introduced a breaking change).
+- [ ] Associated with an issue.
```
Comment (MatFillion) [inline .github/PULL_REQUEST_TEMPLATE.md:43]: Might not need this last one. Associated with an issue.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,31 +0,0 @@
...
-
-### Changed
-* Replaced `DecoratorCommandStrategy` with `DelegatingCommandStrategy`.
-* Replaced `IViewModelView` with `IDispatcher`.
-  * Replaced `IViewModel.View` with `IViewModel.Dispatcher`
-* Update UWP target to 19041 and net framework to 4.7.2
-* Build with VS2022
-* [#09] Update Android target to 10.0
-* [#10] Update the source of the `ViewModelBase` private `ILogger` instance.
-* [#39] Change target Uno.UI version to 4.0.7.
```
Comment (MatFillion) [inline CHANGELOG.md:21]: should we consider dependency updates breaking changes? or at least minor?

Comment (jeanplevesque) [reply CHANGELOG.md:21]: probably minor by default? it's going to be major automatically when it changes the API surface of the lib

File: build/gitversion.yml [hunk 3]
```diff
@@ -1,23 +1,21 @@
...
+next-version: 0.12.0
 continuous-delivery-fallback-tag: ""
+commit-message-incrementing: Enabled
+major-version-bump-message: "^(build|chore|ci|docs|feat|fix|perf|refactor|revert|style|test)(\\([\\w\\s-]*\\))?(!:|:.*\\n\\n((.+\\n)+\\n)?BREAKING CHANGE:\\s.+)"
+minor-version-bump-message: "^(feat)(\\([\\w\\s-]*\\))?:"
+patch-version-bump-message: "^(build|chore|ci|docs|fix|perf|refactor|revert|style|test)(\\([\\w\\s-]*\\))?:"
 increment: none # Disabled as it is not used. Saves time on the GitVersion step
 branches:
   master:
     regex: master
```
Comment (MatFillion) [inline build/gitversion.yml:16]: should we swap to "main" at the same time?

Comment (jeanplevesque) [reply build/gitversion.yml:16]: do we have to?

File: build/gitversion.yml [hunk 4]
```diff
@@ -1,31 +1,29 @@
...
     increment: none
-    source-branches: [master]
   dev:
     regex: dev/.*?/(.*?)
     tag: dev.{BranchName}
     source-branches: [master]
   feature:
     tag: feature.{BranchName}
     regex: feature/(.*?)
-    source-branches: ['master']
```
Comment (vincentcastagna) [inline build/gitversion.yml:28]: Still `master `or should we pass in `main` then as you renamed the branch ?

Comment (jeanplevesque) [reply build/gitversion.yml:28]: I think it points to label at line 15

**Review by MatFillion — APPROVED** [review]

File: build/gitversion.yml (hunk 4 above)
Comment (jeanplevesque) [reply build/gitversion.yml:28]: Anyways we kinda need to test it to see if it works :D

**Pull request merged**